packages/babel-parser/src/parser/statement.js

```diff
@@ -1531,7 +1533,12 @@ export default class StatementParser extends ExpressionParser {
       this.expectOnePlugin(["decorators", "decorators-legacy"]);
       if (this.hasPlugin("decorators")) {
         if (this.getPluginOption("decorators", "decoratorsBeforeExport")) {
-          this.unexpected();
+          this.unexpected(
+            this.state.start,
+            "Decorators must be placed *before* the 'export' keyword." +
+              " You can set the 'decoratorsBeforeExport' option to false to use" +
+              " the 'export @decorator class {}' syntax",
```

Comment:
Maybe we could add where to set the configuration?

> You can set the option on the decorator plugin in your Babel configuration

Reply:
Yeah similar to Logan's question ^ about using the similar error feedback. This error is for Babylon itself which also has plugin options (first time we are doing this), and we'd probably want to do a similar wrapping of the error